test/indexer/test_indexer_controller.py

```diff
@@ -0,0 +1,160 @@
+import json
+import logging
+
+import boto3
+from chalice.app import (
+    SQSRecord,
+)
+import mock
+from more_itertools import (
+    one,
+)
+from moto import (
+    mock_sqs,
+    mock_sts,
+)
+
+from azul import (
+    config,
+    queues,
+)
+from azul.azulclient import (
+    AzulClient,
+)
+from azul.indexer import (
+    BundleFQID,
+)
+from azul.indexer.index_controller import (
+    IndexController,
+)
+from azul.logging import (
+    configure_test_logging,
+)
+from indexer import (
+    IndexerTestCase,
+)
+
+log = logging.getLogger(__name__)
+
+
+# noinspection PyPep8Naming
+def setUpModule():
+    configure_test_logging(log)
+
+
+@mock_sts
+@mock_sqs
+class TestIndexController(IndexerTestCase):
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.index_service.create_indices(self.catalog)
+        self.client = AzulClient()
+        self.controller = IndexController()
+        # noinspection PyPropertyAccess
+        self.controller.index_service = self.index_service
+        self.queue_manager = queues.Queues(delete=True)
+
+    def tearDown(self):
+        self.index_service.delete_indices(self.catalog)
+        super().tearDown()
+
+    def _create_mock_queues(self):
+        sqs = boto3.resource('sqs')
+        for queue_name in config.all_queue_names:
+            sqs.create_queue(QueueName=queue_name)
+
+    def _mock_sqs_record(self, **body):
+        event_dict = {
+            'body': json.dumps(body),
+            'receiptHandle': 'ThisWasARandomString',
+            'attributes': {'ApproximateReceiveCount': 1}
+        }
+        return SQSRecord(event_dict=event_dict, context='controller_test')
+
+    def test_invalid_notification(self):
+        event = [
+            self._mock_sqs_record(action='banana',
+                                  notification='slug',
+                                  catalog=self.catalog)
+        ]
+        self.assertRaises(AssertionError, self.controller.contribute, event)
+
+    def test_remote_reindex(self):
+        event = [self._mock_sqs_record(action='reindex', prefix='ff')]
+        with mock.patch.object(AzulClient, 'do_remote_reindex') as mock_reindex:
+            mock_reindex.return_value = True
+            self.controller.contribute(event)
+            self.assertEqual(1, mock_reindex.call_count)
```

Comment:
There's more you can assert about the mock calls. For reference see how this is handled for `mock_plugin`

Reply:
i dont see much in the repo under `mock_plugin` https://github.com/DataBiosphere/azul/search?q=mock_plugin&type=Code is there a link/file you can direct me to? 

I'll look around for other assertion that are used to mocked objects around the repo.